packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -96,6 +96,7 @@ class HotRunner extends ResidentRunner {
     ReassembleHelper reassembleHelper = _defaultReassembleHelper,
     NativeAssetsBuildRunner? buildRunner,
     required Analytics analytics,
+    this.flavor,
```

Comment:
Consider putting flavor on debugging options instead.

Reply:
This reminded me that the flavor is already available in `debuggingOptions.buildInfo`, so we can just reference that. Addressed in [remove redundant flavor arg from HotRunner](https://github.com/flutter/flutter/pull/132985/commits/03550d0dd9c19673799e0c813c8b129d7fd9fdc9)